Add CloudRay deployment Guide

GeoSegun opened the pull request:
<!-- Thank you for opening a PR! We really appreciate you taking the time to help out 🙌 -->

#### Description (required)

Add CloudRay deployment guide. 
My discord username is @risingsegun

#### Related issues & labels (optional)

- Closes #<!-- Add an issue number if this PR will close it. -->
- Suggested label: <!-- Help us triage by suggesting one of our labels that describes your PR -->

<!-- For a new/changed feature in an upcoming Astro release? -->
<!-- 1. Uncomment the line below, update the minor version number if known, and include a PR link -->
<!-- #### For Astro version: `5.x`. See astro PR [#](url). -->

<!-- 2. Check that your PR includes `<p><Since v="4.x.0" /></p>` and imports the `<Since>` component, if necessary! -->

<!-- #### First-time contributor to Astro Docs? -->

<!-- If you are a member of the Astro Discord, please add your username in the description so we can welcome you there! -->
<!-- https://astro.build/chat -->

ArmandPhilippot commented on public/logos/cloudray.svg:
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<svg version="1.1" xmlns="http://www.w3.org/2000/svg" width="401" height="401">
+<path d="M0 0 C132.33 0 264.66 0 401 0 C401 132.33 401 264.66 401 401 C268.67 401 136.34 401 0 401 C0 268.67 0 136.34 0 0 Z " fill="#551A8B" transform="translate(0,0)"/>
+<path d="M0 0 C0.88486084 0.00563965 1.76972168 0.0112793 2.68139648 0.01708984 C19.26875012 0.32344857 31.7626663 6.45711515 43.3125 18.3125 C50.38620411 26.00337292 56.64666531 37.21088236 56.27734375 47.8671875 C55.72203496 57.33546424 55.72203496 57.33546424 59.25 65.609375 C61.74754881 66.9895134 61.74754881 66.9895134 64.48046875 68.0546875 C74.13857672 72.34421959 80.63224441 82.89663818 8 ...
+</svg>
```
We usually try to ship the smallest number of bytes for logos in the public directory because they don't get processed by Astro. Using [SVGOMG](https://jakearchibald.github.io/svgomg/), I managed to reduce the size from 2.97k to 1.4k. I don't think this affects the way the logo is displayed comparing the two locally:
```suggestion
<svg xmlns="http://www.w3.org/2000/svg" width="401" height="401">
  <path fill="#551A8B" d="M0 0h401v401H0V0Z"/>
  <path fill="#FBFAFC" d="M220.688 119.688c.884.005 1.77.01 2.68.017 16.588.306 29.082 6.44 40.632 18.295 7.074 7.69 13.334 18.898 12.965 29.555-.555 9.468-.555 9.468 2.973 17.742 2.497 1.38 2.497 1.38 5.23 2.445 9.658 4.29 16.152 14.842 20.082 24.25 4.956 14.077 3.545 27.53-2.25 41.008-6.546 13.341-17.421 21.98-31.337 26.812-9.043 2.682-18.032 2.713-27.383 2.689-2.737-.001-5.471.041-8.208.085-1.752.005-3.504.008-5.256.008l-2.459.05c-4.243-.052-5.893-.287-9.369-2.962C217 277 217 277 216.375 273.5c.807-4.52 1.986-5.73 5.625-8.5 3.115-.69 6.145-.627 9.324-.566l2.732-.017a653.43 653.43 0 0 1 8.631.02c16.235.647 16.235.647 31.313-4.437l3.25-1.813c7.651-6.085 11.78-13.601 13.75-23.187.996-10.078-1.264-18.541-7.125-26.813-4.788-5.743-10.981-9.058-17.813-11.875-2.925-1.253-4.216-1.81-6.062-4.312-.277-2.676-.277-2.676-.152-5.883l.084-3.539c.086-2.445.177-4.89.273-7.336.167-7.413-.422-12.806-4.205-19.242l-1.813-3.125C247.779 144.451 241.05 140.35 231 137c-14.038-.862-23.833.14-35.043 9.25-2.155 1.861-2.155 1.861-4.582 4.688C189 153 189 153 185.625 153.5c-3.625-.5-3.625-.5-6.938-2-5.345-2.174-10.32-2.501-15.894-.844-6.12 2.715-10.29 7.121-13.106 13.156-.779 3.613-1.068 7.133-1.246 10.82-.548 2.94-1.324 4.256-3.441 6.368-2.602 1.055-2.602 1.055-5.625 1.875-10.855 3.387-19.48 9.906-24.95 20.043-4.986 10.817-6.382 21.892-2.862 33.457 4.582 11.808 12.294 20.014 23.777 25.293 7.34 2.978 14.204 3.002 22.035 2.832 1.34 0 2.68.003 4.02.008 9.951-.045 9.951-.045 18.3-5.094l1.43-1.789c3.14-3.78 6.403-7.404 9.75-11A1230.364 1230.364 0 0 0 206 230c-1.417-3.252-3.125-5.314-5.688-7.75-2.591-2.527-5.083-5.063-7.437-7.813-4.2-4.895-8.644-9.562-13.077-14.246l-1.786-1.898-1.61-1.702c-2.167-2.46-3.342-4.2-3.84-7.466.438-3.125.438-3.125 2.563-6C178 181 178 181 180.813 180.25c8.706 2.049 14.976 11.723 20.764 18.17 2.29 2.543 4.608 5.06 6.923 7.58l2.71 2.96a1620.69 1620.69 0 0 0 5.892 6.392l1.533 1.66a966.773 966.773 0 0 0 2.926 3.156c4.31 4.672 4.31 4.672 4.751 8.621-.394 4.05-1.664 5.536-4.312 8.586l-2.492 2.887a213.417 213.417 0 0 1-8.625 8.863c-1.8 1.793-3.474 3.639-5.133 5.563-22.028 25.514-22.028 25.514-33.659 26.716-1.409.026-2.818.035-4.228.03l-2.305.015a515.55 515.55 0 0 1-4.808-.003c-2.414-.008-4.826.02-7.24.052-16.593.077-31.171-4.844-43.569-16.361-12.835-14.127-17.653-29.668-17.254-48.469.292-6.044 1.764-11.19 4.313-16.668l1.285-2.82c6.19-12.38 15.489-21.48 28.09-26.993 4.052-1.807 4.052-1.807 6.625-5.18a68.2 68.2 0 0 0 .613-4.515c1.255-8.084 6.817-14.725 12.95-19.77 7.35-5.194 14.173-7.87 23.187-7.847h3.184c2.732.111 5.201.486 7.879 1 2.538.322 3.69.437 5.875-.934 1.12-1.037 2.22-2.099 3.3-3.18 8.525-7.464 19.677-10.147 30.7-10.073Z"/>
</svg>
```

GeoSegun commented:
Hi @ArmandPhilippot  thank you so much for the thoughtful review and kind words! 

I’ve implemented all the suggested changes, and the guide is now updated accordingly. Appreciate the opportunity to contribute 🙌🏽

yanthomasdev commented on src/content/docs/en/guides/deploy/cloudray.mdx:
```diff
@@ -0,0 +1,38 @@
+---
+import { Steps } from '@astrojs/starlight/components';
+
+You can deploy your Astro project using [CloudRay](https://cloudray.io), a centralised platform that helps you manage your servers, organise Bash scripts, and automate deployment tasks across virtual machines and cloud servers.
```
```suggestion
You can deploy your Astro project using [CloudRay](https://cloudray.io), a centralized platform that helps you manage your servers, organize Bash scripts, and automate deployment tasks across virtual machines and cloud servers.
```

yanthomasdev commented on src/content/docs/en/guides/deploy/cloudray.mdx:
```diff
@@ -0,0 +1,38 @@
+
+2. In your CloudRay Dashboard, write a reusable Bash script that clones your Astro repo, installs dependencies, builds your site, and configures a web server. Define any repo-specific values using [CloudRay’s variable groups](https://cloudray.io/docs/variable-groups).
+
+3. Use CloudRay’s Runlog interface to execute your script on your connected server and monitor the deployment in real-time.
```
"real time" as a noun doesn't have hyphen
```suggestion
3. Use CloudRay’s Runlog interface to execute your script on your connected server and monitor the deployment in real time.
```

yanthomasdev reviewed (commented):
Thanks @GeoSegun, I have to small suggestions for you!
